modules/system-disaster-recovery/src/main/java/org/apache/ignite/internal/disaster/system/SystemDisasterRecoveryManagerImpl.java

```diff
@@ -334,6 +356,26 @@ private static boolean isMajorityOfCmgAreSuccesses(
         return successes >= (futuresFromNewCmg.size() + 1) / 2;
     }
 
+    private static String errorMessageForNotEnoughSuccesses(
+            boolean repairMg,
+            Map<String, CompletableFuture<NetworkMessage>> responseFutures
+    ) {
+        if (repairMg) {
+            return String.format(
+                    "Did not get successful response from at least one node, failing cluster reset [failedNode=%s].",
+                    findAnyFailedNodeName(responseFutures)
+            );
+        } else {
+            return "Did not get successful responses from new CMG majority, failing cluster reset.";
+        }
+    }
+
+    private static String findAnyFailedNodeName(Map<String, CompletableFuture<NetworkMessage>> responseFutures) {
+        return responseFutures.entrySet().stream()
+                .filter(entry -> entry.getValue().isCompletedExceptionally())
+                .findAny().orElseThrow().getKey();
```

Comment:
Because of "orElseThrow()" we won't get a weird error?

Reply:
Improved the error message